src/app/view/components/dot-workflow-task-detail/services/dot-workflow-task-detail.service.ts

```diff
@@ -0,0 +1,91 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs/Subject';
+import { Observable } from 'rxjs/Observable';
+import { mergeMap, map, filter } from 'rxjs/operators';
+import { DotMenuService } from '../../../../api/services/dot-menu.service';
+import { of } from 'rxjs/observable/of';
+
+export interface DotEditorAction {
+    header?: string;
+    data: {
+        [key: string]: string;
+    };
+}
+
+/**
+ * Handle the url and events for add and edit workflow tasks components
+ *
+ * @export
+ * @class DotWorkflowTaskDetailService
+ */
+@Injectable()
+export class DotWorkflowTaskDetailService {
+    private data: Subject<DotEditorAction> = new Subject();
+    private _header: Subject<string> = new Subject();
+
+    constructor(private dotMenuService: DotMenuService) {}
+
+    get viewUrl$(): Observable<string> {
+        return this.data.pipe(
+            filter((action: DotEditorAction) => this.isViewUrl(action)),
```

Comment:
You don't need all this, because you only need to handle one url. In fact in this service you don't need ALL this (like the one you use as base), you only need to handle one url and a task is not like a contentlet that we need to add, edit, etc... from the dialog (at least that I know of)